core/sdk-core/src/main/java/software/amazon/awssdk/core/client/handler/BaseClientHandler.java

```diff
@@ -108,6 +120,13 @@ private static SdkHttpFullRequest modifyEndpointHostIfNeeded(SdkHttpFullRequest
                               .build();
     }
 
+    private static void warnOnUnsupportedRequests(SdkHttpFullRequest request) {
+        if (NO_BODY_METHODS.contains(request.method()) && request.contentStreamProvider().isPresent()) {
+            SdkStandardLogger.REQUEST_LOGGER.warn(() -> NO_BODY_METHODS +
+                                                        " requests with bodies are not supported by the default SDK clients.");
+        }
```

Comment:
Is this needed, if we have generator validation?

Reply:
I added this in case someone was using our core with their own plugged in client that supported GET requests with a body. Very unlikely to occur and won't occur for any AWS service. Can just throw or remove here if preferred though.

Reply:
I'd favor throwing or removing. Feels like complexity we're managing for not a lot of gain.